server/src/mds3.init.js

```diff
@@ -657,6 +670,229 @@ function mayAdd_refseq2ensembl(ds, genome) {
 	ds.refseq2ensembl_query = genome.genedb.db.prepare('select ensembl from refseq2ensembl where refseq=?')
 }
 
+/* generate getter as ds.queries.snvindel.byrange.get() when both bcffile(no samples) and maffile are provided
+
+Called to initiate official dataset, in sthis script 
+
+getter input:
+.rglst=[ {chr, start, stop} ] (required)
+.tid2value = { termid1:v1, termid2:v2, ...} (optional, to restrict samples)
+
+getter returns:
+array of variants with array of sample ids attached to each variant
+as .samples=[ {sample_id} ]
+*/
+export async function snvindelByRangeGetter_bcfMaf(ds, genome) {
+	const q = ds.queries.snvindel.byrange
+	/* q{}
+	._tk={}
+        	.file= absolute path to bcf file
+		.maffile= absolute path to maf file
+		.dir=str
+		.nochr=true
+		.sampleIdx=int // index of sample column in maf file
+		.formatIdx=Map()
+		.info={}
+			<ID>: {}
+				ID: 'CSQ',
+				Number: '.',
+				Type: 'String',
+				Description: 'Cons
+		.format={} // null if no format
+			<ID>: {}
+				ID: 'dna_assay',
+				Number: '1',
+				Type: 'String',
+				Description: '...'
+		.samples=[ {name:str}, ... ]
+	.
+	*/
+	await utils.init_one_vcfMaf(q._tk, genome, true) // "true" to indicate file is bcf but not vcf
+	// q._tk{} is initiated
+	if (q._tk.format) {
+		for (const id in q._tk.format) {
+			if (id == 'GT') {
+				q._tk.format[id].isGT = true
+			}
+		}
+	}
+	if (ds.queries.snvindel.format4filters) {
+		// array of format keys; allow to be used as filters on client
+		if (!Array.isArray(ds.queries.snvindel.format4filters)) throw 'snvindel.format4filters[] is not array'
+		for (const k of ds.queries.snvindel.format4filters) {
+			if (q._tk.format[k]) q._tk.format[k].isFilter = true // allow it to work as filter on client
+		}
+		delete ds.queries.snvindel.format4filters
+	}
+	if (q._tk.info) {
+		if (q.infoFields) {
```

Comment:
i don't see `infoFields` used in any ds

Reply:
it comes from the function "snvindelByRangeGetter_bcf". I kept it there in case it is useful. Please also check line 1001